cmd/soroban-cli/src/commands/contract/deploy/wasm.rs

```diff
@@ -218,25 +228,43 @@ impl NetworkRunnable for Cmd {
             salt,
             &key,
         )?;
+
         if self.fee.build_only {
+            output.check("Done!");
             return Ok(TxnResult::Txn(txn));
         }
 
+        output.info("Simulating deploy transaction…");
+
         let txn = client.simulate_and_assemble_transaction(&txn).await?;
         let txn = self.fee.apply_to_assembled_txn(txn).transaction().clone();
+
         if self.fee.sim_only {
+            output.check("Done!");
             return Ok(TxnResult::Txn(txn));
         }
+
+        output.globe("Submitting deploy transaction…");
+        utils::log_transaction(&output, &txn, &network, true)?;
+
         let get_txn_resp = client
             .send_transaction_polling(&config.sign_with_local_key(txn).await?)
             .await?
             .try_into()?;
+
         if global_args.map_or(true, |a| !a.no_cache) {
             data::write(get_txn_resp, &network.rpc_uri()?)?;
         }
```

Comment:
This has always had a bad code smell. With a refactor we can ensure that the global is no longer an Option.